### PR TITLE
OMB policy router update for frontend

### PR DESCRIPTION
**Pull request by rememberlenny**
# CONTEXT 

https://github.com/18F/omb-eregs/issues/551

### CODE REVIEW

File: ui/util/api/queries.js [hunk 1]
```diff
@@ -30,10 +30,11 @@ export async function policiesData({ query }) {
     endpoints.policies.withIds(query.id__in),
     endpoints.topics.withIds(query.requirements__topics__id__in),
     endpoints.policies.fetch(Object.assign({ ordering: 'policy_number' }, query)),
+    endpoints.policies.withOmbPolicyIds(query.id__in),
```
Comment (cmc333333) [inline ui/util/api/queries.js]: I don't think this'll get you the results you're looking for. There were five different queries happenings here, four of which were looking up ids (to grab names to display to the user) but the fifth (`policies.fetch`) grabbed a filtered list of policies, depending on the user's query. By adding this line, we're extending the lookup search (i.e., we can find more policy _names_) but not the policy search (i.e. we won't get additional matches).

Minor: if we're keeping this, it'd probably make sense to keep the id-lookup queries together (meaning we should move this line up).

File: ui/util/api/queries.js [hunk 2]
```diff
@@ -144,10 +145,15 @@ async function propagate404(fn) {
 
 export async function policyData({ query }) {
   const reqQuery = {
-    policy_id: query.policyId,
     page: query.page || '1',
   };
 
+  if (isNaN(query.policyId) === true) {
+    reqQuery.omb_policy_id = query.policyId;
```
[the remark this answers is not in the document]
Comment (rememberlenny) [reply ui/util/api/queries.js]: Ah ha.

[the remark this answers is not in the document]
Comment (rememberlenny) [reply ui/util/api/queries.js]: You are absolutely right. Fixing.

File: ui/util/api/queries.js [hunk 3]
```diff
@@ -30,10 +30,11 @@ export async function policiesData({ query }) {
     endpoints.policies.withIds(query.id__in),
     endpoints.topics.withIds(query.requirements__topics__id__in),
     endpoints.policies.fetch(Object.assign({ ordering: 'policy_number' }, query)),
+    endpoints.policies.withOmbPolicyIds(query.id__in),
   ]);
   return {
     existingAgencies: results[0],
-    existingPolicies: results[1],
+    existingPolicies: results[4].length > 0 ? results[4] : results[1],
```
[the remark this answers is not in the document]
Comment (rememberlenny) [reply ui/util/api/queries.js]: Updated!

File: ui/util/api/queries.js (hunk 1 above)
Comment (rememberlenny) [reply ui/util/api/queries.js]: I missed this. Fixing.

File: ui/util/api/endpoints.js [hunk 4]
```diff
@@ -32,13 +32,23 @@ export class Endpoint {
     return this.fetch(params).then(data => data.results);
   }
 
-  withIds(idString) {
+  queryWithParamType(paramType, idString) {
     if (idString) {
-      return this.fetchResults({ id__in: idString });
+      const fetchParam = {};
+      fetchParam[paramType] = idString;
+      return this.fetchResults(fetchParam);
```
[the remark this answers is not in the document]
Comment (rememberlenny) [reply ui/util/api/endpoints.js]: I knew something like this was possible, but I couldnt pinpoint the syntax. Thanks.

**Comment by rememberlenny** [comment]
@cmc333333 Fixed.

**Comment by cmc333333** [comment]
@rememberlenny added a suggested change in #604, but I think we're largely good to go.

**Pull request merged**